app/src/main/java/io/pslab/fragment/SoundMeterDataFragment.java

```diff
@@ -0,0 +1,365 @@
+     **********************************************************************************************
+     */
+    private void updateMeter(short[] buffer) {
+        double sqrsum = 0.0;
+        for (int i = 0; i < buffer.length; ++i) {
+            sqrsum += Math.pow(buffer[i],2);
+        }
+        double rmsamp = Math.sqrt((sqrsum/buffer.length));
+
+        maxRmsAmp = Math.max(rmsamp,maxRmsAmp);
+        minRmsAmp = Math.min(rmsamp,minRmsAmp);
+        rmsSum = (count<Integer.MAX_VALUE)?(rmsSum+rmsamp):rmsamp;
+        count = (count<Integer.MAX_VALUE)?(count+1):1;
+        double avgRmsAmp = rmsSum/count;
+
+        double loudness = rmsamp>0?(10*Math.log10(rmsamp/1d)):1;
+        double maxLoudness = maxRmsAmp>0?(10*Math.log10(maxRmsAmp/1d)):1;
+        double minLoudness = minRmsAmp>0?(10*Math.log10(minRmsAmp/1d)):1;
+        double avgLoudness = avgRmsAmp>0?(10*Math.log10(avgRmsAmp/1d)):1;
+
+        decibelMeter.setSpeedAt((float)loudness);
+        statMax.setText(String.format(Locale.getDefault(), PSLabSensor.SOUNDMETER_DATA_FORMAT,maxLoudness));
+        statMin.setText(String.format(Locale.getDefault(), PSLabSensor.SOUNDMETER_DATA_FORMAT,minLoudness));
+        statMean.setText(String.format(Locale.getDefault(), PSLabSensor.SOUNDMETER_DATA_FORMAT,avgLoudness));
+        writeLog(System.currentTimeMillis(),(float)loudness);
+    }
+
+    private void updateChart(short[] buffer) {
+        for (int i = 0; i < buffer.length; ++i) {
+            float x = (System.currentTimeMillis() - recordStartTime)/1000f;
+            float y = buffer[i]/1000f;
+            if(chartQ.size()>=buffer.length)
+                chartQ.removeFirst();
+            chartQ.addLast(new Entry(x,y));
+            Log.i(TAG,"x : "+x+"  "+"y : "+y);
+        }
+        List<Entry> entries = new ArrayList<>(chartQ);
+        LineDataSet dataSet = new LineDataSet(entries, "Amplitude");
+        dataSet.setDrawCircles(false);
+        dataSet.setDrawValues(false);
+        dataSet.setLineWidth(0.5f);
+        mChart.setData(new LineData(dataSet));
+        mChart.notifyDataSetChanged();
+        mChart.setVisibleXRangeMaximum(entries.size());
+        mChart.invalidate();
+    }
+
+    /**
+     * Method to play data which was previously recorded
+     */
+    public void playData() {
+        CustomSnackBar.showSnackBar(getActivity().findViewById(android.R.id.content),"This feature is yet to be implemented",
+                null,null,Snackbar.LENGTH_SHORT);
+        //TODO: To be implemented
+    }
+
+    /**
+     * Method to stop playing the previously recorded data
+     */
+    public void stopData() {
+        CustomSnackBar.showSnackBar(getActivity().findViewById(android.R.id.content),"This feature is yet to be implemented",
+                null,null,Snackbar.LENGTH_SHORT);
+        //TODO: To be implemented
+    }
+
+    /**********************************************************************************************
```

Comment:
Use a consistent commenting style

Reply:
Thanks for the feedback! However i think, i did not completely understand what u meant by a consistent style. It would be really helpful if you could suggest me the type of style I should use for the further changes.

Reply:
Hi, I would say as in line 283. Also do not hardcode strings. Use them from `strings.xml`

Reply:
okay! you mean the **This_ feature is yet to be implemented** string right ?

Reply:
Yes that part ..

Reply:
Okay cool, I will do the needful.